src/handle.rs

```diff
@@ -160,14 +173,15 @@ impl<T> Clone for Global<T> {
 
 impl<T> Drop for Global<T> {
   fn drop(&mut self) {
-    unsafe {
-      if self.isolate_handle.get_isolate_ptr().is_null() {
-        // This `Global` handle is associated with an `Isolate` that has already
-        // been disposed.
-      } else {
-        // Destroy the storage cell that contains the contents of this Global.
-        v8__Global__Reset(self.data.cast().as_ptr())
+    let HandleInfo { data, host } = self.get_handle_info();
+    match host {
+      // This `Global` handle is associated with an `Isolate` that has already
+      // been disposed.
+      HandleHost::DisposedIsolate => {}
+      HandleHost::UnlockedIsolate => {
+        panic!("Global dropped outside of locked isolate context")
```

Comment:
Not that I necessarily see a better way, but this seems unfortunate.

Reply:
I added a disposal bin for dropped global's that `reset`s and drops the v8 references the next time the isolate is entered/exited or after the Isolate is dropped. I don't know if this is a better approach, but I thought it complements the general flexibility of the Global type. It removes the mentioned error and eventually cleans up the reference.

The disposal list is tucked away in the `IsolateAnnex`.

Additionally, the `Global` handles hold a `Weak<IsolateAnnex>` reference (vs. cloned IsolateHandle arc) so the Isolate doesn't get held alive by dangling handles. It uses this to enqueue itself to the disposal queue if the global was dropped outside of the locked isolate scope.